### DISCUSSION TITLE
GitLab not automerging mr

### DISCUSSION THREAD

**Discussion by romainjrd** [renovatebot/renovate#28153] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

Hello,

I have a bug using Renovate on self-hosted GitLab [v16.5.8](https://gitlab.com/gitlab-org/gitlab-foss/-/tags/v16.5.8).

In a nutshell, I'm trying to update FROM image in Dockerfiles stored in the same GitLab repository.

In the Renovate config, I activated `automerge` and `platformAutomerge`. I'm **grouping updates per Dockerfile**, this way I have one Merge Request concerning one Dockerfile. As a result, I'm avoiding conflicts. 

I also configured my GitLab repository to `Merge commit`. It allows Renovate to Merge multiple branches without rebasing.

When running Renovate, branches are **correctly** created, tests are **executed** and `platformAutomerge` is set.

However, after tests are finished, some Merge Requests are still open. Tests are successful and the Merge Request is `set to be merged automatically when the pipeline succeeds`.

![image](https://github.com/renovatebot/renovate/assets/155526382/d8023d98-d8ef-43fd-9eae-05a60e054f2a)

The Merge Request should be merged because there are no conflicts. 

Has anyone ever had this bug ? Did I miss something in the configuration ?

Many thanks !

renovate.json :
```json
{
  "extends": [
  ":dependencyDashboard",
  ":semanticPrefixFixDepsChoreOthers",
  ":automergePr",
  ":automergeRequireAllStatusChecks",
  ":disableMajorUpdates",
  "replacements:all",
  "workarounds:all"
  ],
  "platformAutomerge": true,
  "automerge": true,
  "rebaseWhen": "conflicted",
  "prConcurrentLimit": 0,
  "prHourlyLimit": 0,
  "branchConcurrentLimit": 0,
  "baseBranches": ["main"],
  "enabledManagers": ["dockerfile"],
  "recreateWhen": "always",
  "additionalBranchPrefix": "{{baseDir}}-",
  "packageRules": [
    {
      "matchManagers": ["dockerfile"],
      "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(-(?<compatibility>.*))?$"
    },
    {
      "matchManagers": ["dockerfile", "regex"],
      "enabled": true,
      "groupName": "Dockerfiles changes",
      "groupSlug": "dockerfile"
    },
    {
      "matchManagers": ["dockerfile"],
      "enabled": false,
      "matchUpdateTypes": [
        "major"
      ]
    }
  ]
}
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
This is not a bug. Based on your screenshot, Renovate has set the automerge flag in GitLab as needed and the rest is up to GitLab, not Renovate
